components/raftstore/src/store/worker/read.rs

```diff
@@ -813,6 +804,12 @@ where
             return Ok(None);
         }
 
+        // Check witness
+        if find_peer_by_id(&delegate.region, delegate.peer_id).map_or(true, |p| p.is_witness) {
```

Comment:
Any test case?

Reply:
It's very straightforward to avoid the unwrap panic, don't think it's necessary

Reply:
If it's that straightforward, why it's not avoided in the last PR?

Reply:
Why is the default true?

Reply:
Not sure whether it is a witness, better reject it

Reply:
Technically if epoch matches then it must be searchable. So `unwrap` is actually OK here.